<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Genetic colocalisation • gwasglue</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Genetic colocalisation">
<meta property="og:description" content="gwasglue">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gwasglue</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/cojo.html">Conditional analysis of VCF files</a>
    </li>
    <li>
      <a href="../articles/colocalisation.html">Genetic colocalisation</a>
    </li>
    <li>
      <a href="../articles/finemapping.html">Clumping and finemapping</a>
    </li>
    <li>
      <a href="../articles/gwas2020.html">Major changes to the IEU GWAS resources for 2020</a>
    </li>
    <li>
      <a href="../articles/mr.html">Mendelian randomization</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/mrcieu/gwasglue/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Genetic colocalisation</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/mrcieu/gwasglue/blob/master/vignettes/colocalisation.Rmd"><code>vignettes/colocalisation.Rmd</code></a></small>
      <div class="hidden name"><code>colocalisation.Rmd</code></div>

    </div>

    
    
<p>Here we’ll perform colocalisation analysis for a particular region, and plot the regions as well. We’ll do the same analysis two ways:</p>
<ul>
<li>querying the association data from the <a href="https://gwas.mrcieu.ac.uk/">IEU GWAS database</a>, and</li>
<li>downloading the <a href="https://github.com/MRCIEU/gwas_vcf_spec">GWAS VCF</a> files and querying those.</li>
</ul>
<p>We’ll use the example of LDL cholesterol <a href="https://gwas.mrcieu.ac.uk/datasets/ieu-a-300/">ieu-a-300</a> and coronary heart disease <a href="https://gwas.mrcieu.ac.uk/datasets/ieu-a-7/">ieu-a-7</a>.</p>
<p>Load libraries:</p>
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/warning.html">suppressWarnings</a></span>({
    <span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">gwasglue</span>)
    <span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">dplyr</span>)
    <span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">gassocplot</span>)
    <span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">coloc</span>)
}))</pre></body></html></div>
<div id="ieugwasr" class="section level2">
<h2 class="hasAnchor">
<a href="#ieugwasr" class="anchor"></a>ieugwasr</h2>
<p>First find a region that we know to be associated with LDL cholesterol.</p>
<div class="sourceCode" id="cb2"><html><body><pre class="r"><span class="no">top</span> <span class="kw">&lt;-</span> <span class="kw pkg">ieugwasr</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/ieugwasr/man/tophits.html">tophits</a></span>(<span class="st">'ieu-a-300'</span>) <span class="kw">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html">arrange</a></span>(<span class="no">p</span>)
<span class="co">#&gt; API: public: http://gwas-api.mrcieu.ac.uk/</span>
<span class="no">top</span>
<span class="co">#&gt; # A tibble: 81 x 12</span>
<span class="co">#&gt;       beta position     se         p chr        n id    rsid  ea    nea      eaf</span>
<span class="co">#&gt;      &lt;dbl&gt;    &lt;int&gt;  &lt;dbl&gt;     &lt;dbl&gt; &lt;chr&gt;  &lt;int&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;  &lt;dbl&gt;</span>
<span class="co">#&gt;  1 -0.221    1.12e7 0.0061 1.00e-200 19    170608 ieu-… rs65… T     G     0.0976</span>
<span class="co">#&gt;  2 -0.485    4.54e7 0.0119 1.00e-200 19    139198 ieu-… rs72… A     G     0.0317</span>
<span class="co">#&gt;  3 -0.150    1.10e8 0.0047 1.00e-200 1     172825 ieu-… rs49… T     C     0.187 </span>
<span class="co">#&gt;  4  0.119    2.13e7 0.004  9.48e-183 2     173007 ieu-… rs13… A     G     0.288 </span>
<span class="co">#&gt;  5 -0.497    5.55e7 0.018  8.57e-143 1      77417 ieu-… rs11… T     G     0.0172</span>
<span class="co">#&gt;  6 -0.446    4.54e7 0.0183 2.99e-122 19     99409 ieu-… rs12… A     G     0.0172</span>
<span class="co">#&gt;  7 -0.0806   4.41e7 0.0041 4.84e- 83 2     172940 ieu-… rs65… C     T     0.706 </span>
<span class="co">#&gt;  8  0.0733   7.47e7 0.0038 7.79e- 78 5     168357 ieu-… rs12… C     T     0.431 </span>
<span class="co">#&gt;  9 -0.189    4.52e7 0.0112 7.29e- 62 19    170260 ieu-… rs29… C     T     0.0211</span>
<span class="co">#&gt; 10 -0.0564   1.26e8 0.0036 2.10e- 50 8     172963 ieu-… rs29… T     A     0.468 </span>
<span class="co">#&gt; # … with 71 more rows, and 1 more variable: trait &lt;chr&gt;</span></pre></body></html></div>
<p>Choose the best signal and create a range</p>
<div class="sourceCode" id="cb3"><html><body><pre class="r"><span class="no">chrpos</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="no">top</span>$<span class="no">chr</span>[<span class="fl">1</span>], <span class="st">":"</span>, <span class="no">top</span>$<span class="no">position</span>[<span class="fl">1</span>] - <span class="fl">90000</span>, <span class="st">"-"</span>, <span class="no">top</span>$<span class="no">position</span>[<span class="fl">1</span>] + <span class="fl">90000</span>)
<span class="no">chrpos</span>
<span class="co">#&gt; [1] "19:11112306-11292306"</span></pre></body></html></div>
<p>Extract, harmonise and format the data</p>
<div class="sourceCode" id="cb4"><html><body><pre class="r"><span class="no">out</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/ieugwasr_to_coloc.html">ieugwasr_to_coloc</a></span>(<span class="kw">id1</span><span class="kw">=</span><span class="st">'ieu-a-300'</span>, <span class="kw">id2</span><span class="kw">=</span><span class="st">'ieu-a-7'</span>, <span class="kw">chrompos</span><span class="kw">=</span><span class="no">chrpos</span>)
<span class="co">#&gt; Type information not available for ieu-a-300. Assuming 'quant' but override using 'type' arguments.</span>
<span class="co">#&gt; Type information not available for ieu-a-7. Assuming 'quant' but override using 'type' arguments.</span></pre></body></html></div>
<p>Run colocalisation analysis</p>
<div class="sourceCode" id="cb5"><html><body><pre class="r"><span class="no">res</span> <span class="kw">&lt;-</span> <span class="kw pkg">coloc</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/coloc/man/coloc.abf.html">coloc.abf</a></span>(<span class="no">out</span><span class="kw">[[</span><span class="fl">1</span>]], <span class="no">out</span><span class="kw">[[</span><span class="fl">2</span>]])
<span class="co">#&gt; Warning in sdY.est(d$varbeta, d$MAF, d$N): estimating sdY from maf and varbeta,</span>
<span class="co">#&gt; please directly supply sdY if known</span>

<span class="co">#&gt; Warning in sdY.est(d$varbeta, d$MAF, d$N): estimating sdY from maf and varbeta,</span>
<span class="co">#&gt; please directly supply sdY if known</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt; 3.85e-289  3.38e-10 7.35e-281  6.35e-02  9.36e-01 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 93.6%"</span></pre></body></html></div>
<p>Plot</p>
<div class="sourceCode" id="cb6"><html><body><pre class="r"><span class="no">temp</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/coloc_to_gassocplot.html">coloc_to_gassocplot</a></span>(<span class="no">out</span>)
<span class="co">#&gt; Extracting LD matrix for 148 variants</span>
<span class="co">#&gt; Please look at vignettes for options on running this locally if you need to run many instances of this command.</span>
<span class="co">#&gt; Warning in ieugwasr::ld_matrix(markers[["marker"]], with_alleles = FALSE, : The following variants are not present in the LD reference panel</span>
<span class="co">#&gt; rs17001200</span>
<span class="co">#&gt; rs2738459</span>
<span class="co">#&gt; rs8110695</span>
<span class="co">#&gt; Found 145 variants in LD reference panel</span>
<span class="kw pkg">gassocplot</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/gassocplot/man/stack_assoc_plot.html">stack_assoc_plot</a></span>(<span class="no">temp</span>$<span class="no">markers</span>, <span class="no">temp</span>$<span class="no">z</span>, <span class="no">temp</span>$<span class="no">corr</span>, <span class="kw">traits</span><span class="kw">=</span><span class="no">temp</span>$<span class="no">traits</span>)</pre></body></html></div>
<p><img src="colocalisation_files/figure-html/unnamed-chunk-7-1.png" width="700"></p>
<pre><code>#&gt; TableGrob (2 x 1) "arrange": 2 grobs
#&gt;   z     cells    name              grob
#&gt; 1 1 (1-1,1-1) arrange    gtable[layout]
#&gt; 2 2 (2-2,1-1) arrange gtable[guide-box]</code></pre>
</div>
<div id="gwasvcf" class="section level2">
<h2 class="hasAnchor">
<a href="#gwasvcf" class="anchor"></a>gwasvcf</h2>
<p>Let’s do the same with the vcf files (and the indexes). Download from here:</p>
<ul>
<li><a href="https://gwas.mrcieu.ac.uk/files/ieu-a-300/ieu-a-300.vcf.gz">https://gwas.mrcieu.ac.uk/files/ieu-a-300/ieu-a-300.vcf.gz</a></li>
<li><a href="https://gwas.mrcieu.ac.uk/files/ieu-a-300/ieu-a-300.vcf.gz.tbi">https://gwas.mrcieu.ac.uk/files/ieu-a-300/ieu-a-300.vcf.gz.tbi</a></li>
<li><a href="https://gwas.mrcieu.ac.uk/files/ieu-a-7/ieu-a-7.vcf.gz">https://gwas.mrcieu.ac.uk/files/ieu-a-7/ieu-a-7.vcf.gz</a></li>
<li><a href="https://gwas.mrcieu.ac.uk/files/ieu-a-7/ieu-a-7.vcf.gz.tbi">https://gwas.mrcieu.ac.uk/files/ieu-a-7/ieu-a-7.vcf.gz.tbi</a></li>
</ul>
<p>Set a region to plot:</p>
<div class="sourceCode" id="cb8"><html><body><pre class="r"><span class="no">chrpos</span> <span class="kw">&lt;-</span> <span class="st">"19:11112306-11292306"</span></pre></body></html></div>
<p>Extract region from vcf files and convert to coloc object</p>
<div class="sourceCode" id="cb9"><html><body><pre class="r"><span class="no">vout</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/gwasvcf_to_coloc.html">gwasvcf_to_coloc</a></span>(<span class="st">"ieu-a-300.vcf.gz"</span>, <span class="st">"ieu-a-7.vcf.gz"</span>, <span class="no">chrpos</span>)
<span class="co">#&gt; Warning: multiple methods tables found for 'type'</span>
<span class="co">#&gt; Warning: replacing previous import 'BiocGenerics::type' by 'DelayedArray::type'</span>
<span class="co">#&gt; when loading 'SummarizedExperiment'</span>
<span class="co">#&gt; 'tools_bcftools' option is not set, using native read which may be substantially slower. See 'set_bcftools' for information.</span>
<span class="co">#&gt; 'tools_bcftools' option is not set, using native read which may be substantially slower. See 'set_bcftools' for information.</span></pre></body></html></div>
<p>Run colocalisation analysis</p>
<div class="sourceCode" id="cb10"><html><body><pre class="r"><span class="no">vres</span> <span class="kw">&lt;-</span> <span class="kw pkg">coloc</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/coloc/man/coloc.abf.html">coloc.abf</a></span>(<span class="no">vout</span><span class="kw">[[</span><span class="fl">1</span>]], <span class="no">vout</span><span class="kw">[[</span><span class="fl">2</span>]])
<span class="co">#&gt; Warning in sdY.est(d$varbeta, d$MAF, d$N): estimating sdY from maf and varbeta,</span>
<span class="co">#&gt; please directly supply sdY if known</span>

<span class="co">#&gt; Warning in sdY.est(d$varbeta, d$MAF, d$N): estimating sdY from maf and varbeta,</span>
<span class="co">#&gt; please directly supply sdY if known</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt; 3.85e-289  3.38e-10 7.35e-281  6.35e-02  9.36e-01 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 93.6%"</span></pre></body></html></div>
<p>Plot</p>
<div class="sourceCode" id="cb11"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">gassocplot</span>)
<span class="no">temp</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/coloc_to_gassocplot.html">coloc_to_gassocplot</a></span>(<span class="no">vout</span>)
<span class="co">#&gt; Extracting LD matrix for 148 variants</span>
<span class="co">#&gt; Please look at vignettes for options on running this locally if you need to run many instances of this command.</span>
<span class="co">#&gt; Warning in ieugwasr::ld_matrix(markers[["marker"]], with_alleles = FALSE, : The following variants are not present in the LD reference panel</span>
<span class="co">#&gt; rs8110695</span>
<span class="co">#&gt; rs2738459</span>
<span class="co">#&gt; rs17001200</span>
<span class="co">#&gt; Found 145 variants in LD reference panel</span>
<span class="kw pkg">gassocplot</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/gassocplot/man/stack_assoc_plot.html">stack_assoc_plot</a></span>(<span class="no">temp</span>$<span class="no">markers</span>, <span class="no">temp</span>$<span class="no">z</span>, <span class="no">temp</span>$<span class="no">corr</span>, <span class="kw">traits</span><span class="kw">=</span><span class="no">temp</span>$<span class="no">traits</span>)</pre></body></html></div>
<p><img src="colocalisation_files/figure-html/unnamed-chunk-11-1.png" width="700"></p>
<pre><code>#&gt; TableGrob (2 x 1) "arrange": 2 grobs
#&gt;   z     cells    name              grob
#&gt; 1 1 (1-1,1-1) arrange    gtable[layout]
#&gt; 2 2 (2-2,1-1) arrange gtable[guide-box]</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Gibran Hemani.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
